pkg/storage/fs/cephfs/cephfs.go

```diff
@@ -615,18 +624,205 @@ func (fs *cephfs) UpdateStorageSpace(ctx context.Context, req *provider.UpdateSt
 	return nil, errtypes.NotSupported("unimplemented")
 }
 
+var fnvHash hash.Hash32 = fnv.New32a()
+
+func getHash(s string) uint64 {
+	fnvHash.Write([]byte(s))
+	defer fnvHash.Reset()
+	buf := bytes.NewReader(fnvHash.Sum(nil))
+	var res uint32
+	binary.Read(buf, binary.BigEndian, &res)
+	return uint64(res)
+	// h := fnvHash.Sum(nil)
+	// return uint64(h[0]) + uint64(h[1])*uint64(2<<8) + uint64(h[2])*uint64(2<<16) + uint64(h[3])*uint64(2<<24)
+}
+
+func encodeLock(l *provider.Lock) string {
+	data, _ := json.Marshal(l)
+	return b64.StdEncoding.EncodeToString(data)
+}
+
+func decodeLock(content string) (*provider.Lock, error) {
+	d, err := b64.StdEncoding.DecodeString(content)
+	if err != nil {
+		return nil, err
+	}
+
+	l := new(provider.Lock)
+	err = json.Unmarshal(d, l)
+	if err != nil {
+		return nil, err
+	}
+
+	return l, nil
+}
+
+// TODO(lopresti) part of this logic is duplicated from eosfs.go, should be factored out
```

Comment:
That's a good idea, yes, but e.g. the `RefreshLock` method is entirely based on the others (in a way it's not "primitive") and it could well be implemented at the storage provider level.